src/d3a/models/strategy/home_meter.py

```diff
@@ -0,0 +1,577 @@
+"""
+Copyright 2018 Grid Singularity
+This file is part of D3A.
+This program is free software: you can redistribute it and/or modify it under the terms of the GNU
+General Public License as published by the Free Software Foundation, either version 3 of the
+License, or (at your option) any later version.
+
+This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+General Public License for more details.
+
+You should have received a copy of the GNU General Public License along with this program. If not,
+see <http://www.gnu.org/licenses/>.
+"""
+from logging import getLogger
+from pathlib import Path
+from typing import Dict, Union
+
+from d3a_interface.constants_limits import ConstSettings
+from d3a_interface.device_validator import HomeMeterValidator
+from d3a_interface.read_user_profile import read_arbitrary_profile, InputProfileTypes
+from d3a_interface.utils import find_object_of_same_weekday_and_time
+from numpy import random
+from pendulum import duration
+
+from d3a import constants
+from d3a.constants import FLOATING_POINT_TOLERANCE, DEFAULT_PRECISION
+from d3a.d3a_core.exceptions import D3AException
+from d3a.d3a_core.exceptions import MarketException
+from d3a.d3a_core.util import get_market_maker_rate_from_config
+from d3a.models.market import Market
+from d3a.models.market.market_structures import Offer
+from d3a.models.state import HomeMeterState
+from d3a.models.strategy import BidEnabledStrategy
+from d3a.models.strategy.update_frequency import (
+    TemplateStrategyBidUpdater, TemplateStrategyOfferUpdater)
+
+log = getLogger(__name__)
+
+
+# pylint: disable=invalid-name
```

Comment:
Pylint will complain about all cases in which we use `Wh` in variable names. For example:
> Variable name "offer_energy_kWh" doesn't conform to snake_case naming style (380:8) [invalid-name]

So your suggestion of adding it to `.pylintrc` makes sense, since this is widely used in our code.